pkg/server/handlers.go

```diff
@@ -11,10 +13,51 @@ import (
 	"github.com/grafana/iot-sitewise-datasource/pkg/models"
 )
 
+func getErrorResponse(err error, message string) *backend.QueryDataResponse {
+	resp := backend.Responses{
+		"A": backend.DataResponse{
+			Error: errors.New(message + ": " + err.Error()),
+		},
+	}
+	return &backend.QueryDataResponse{Responses: resp}
+}
+
 func processQueries(ctx context.Context, req *backend.QueryDataRequest, handler QueryHandlerFunc) *backend.QueryDataResponse {
 	res := backend.Responses{}
-	for _, v := range req.Queries {
-		res[v.RefID] = handler(ctx, req, v)
+
+	var datasourceJsonData struct {
+		DefaultRegion string `json:"defaultRegion"`
+	}
+
+	defaultRegion := ""
+	if req.PluginContext.DataSourceInstanceSettings != nil &&
+		req.PluginContext.DataSourceInstanceSettings.JSONData != nil {
+
+		if err := json.Unmarshal(req.PluginContext.DataSourceInstanceSettings.JSONData, &datasourceJsonData); err != nil {
+			return getErrorResponse(err, "failed to unmarshal datasource JSON data")
+		}
+		if datasourceJsonData.DefaultRegion != "" {
+			defaultRegion = datasourceJsonData.DefaultRegion
+		}
+	}
+
+	for _, q := range req.Queries {
+		var queryJSON map[string]interface{}
+
+		if err := json.Unmarshal(q.JSON, &queryJSON); err != nil {
```

Comment:
nit: I don't really love having to unmarshal and re-marshal the query (doing this can be a little expensive and we presumably need to unmarshal it again) but I do get doing the suboptimal thing to get this through)

Reply:
Managed to avoid it by applying Kevin's suggestion https://github.com/grafana/iot-sitewise-datasource/pull/556#discussion_r2273593932